echo.go

```diff
@@ -972,6 +974,18 @@ func handlerName(h HandlerFunc) string {
 	return t.String()
 }
 
+func (e *Echo) hasValidHandlerAndMiddleware(handler HandlerFunc, middlewares ...MiddlewareFunc) bool {
+	if handler == nil {
+		for _, middleware := range middlewares {
+			if middleware != nil {
+				return true
+			}
+		}
+		return false
```

Comment:
You mean when both of them were nil ?
```suggestion
		panic("can not add route with nil handler and middlewares")
```